js/reference.js

```diff
@@ -6595,3 +6618,126 @@
             }
         });
     }
+
+    /**
+     *
+     * @param {ERMrest.Reference} reference reference for the association table
+     * @param {Object} prefillObject computed prefill object after extracting the query param and fetching the data from cookie storage
+     * @param {Object} prefillAnnotation
+     */
+    function Prefill (reference, prefillObject, prefillAnnotation) {
```

Comment:
You need to clean this up. It's really hard to understand what attributes this prototype has. Please follow what we've done in other places in ermrestjs. Populate the private props inside the constructor and then add getter function for any one that you want to expose.

```js
function Prefill(reference, prefillObject) {
   // populate all the _NAME props here, for example:
  this._mainColumn = ...
}

Prefill.prototype = {
  get mainColumn() {
    return this._mainColumn
 }
  
}
```

And do could do the following for props that have their own logic that doesn't involve other props:

```
get andFiltersForLeaf () {
  if (this._andFiltersForLeaf === undefined) {
    ...
    this._andFiltersForLeaf = ...
  }
  return this._andFiltersForLeaf;
}

```